system/src/system_control_internal.cpp

```diff
@@ -357,6 +357,10 @@ void SystemControl::processRequest(ctrl_request* req, ControlRequestChannel* /*
         setResult(req, ctrl::wifi::scanNetworks(req));
         break;
     }
+    case CTRL_REQUEST_WIFI_MAC_ADDRESS: {
```

Comment:
This request should not really be necessary. MAC addresses are available since introduction of `CTRL_REQUEST_NETWORK_GET_INTERFACE_LIST` and `CTRL_REQUEST_NETWORK_GET_INTERFACE`: https://github.com/particle-iot/device-os-protobuf/blob/main/control/network.proto#L94

Reply:
We could add a new request to get a 'base MAC' and size of the allocation per device, but it's not going to be wifi-specific, it's more of a device identifier-like data similar to serial/device id/mobile secret etc.

For Gen 4 devices the size is 4 for example:
0. WiFi STA
1. WiFi AP
2. BLE
3. Ethernet